fn.go

```diff
@@ -20,34 +32,204 @@ type Function struct {
 	log logging.Logger
 }
 
+const (
+	errFmtInvalidFunction   = "invalid function input: %s"
+	errFmtInvalidReadyValue = "%s is invalid, ready annotation must be True, Unspecified, or False"
+	errFmtInvalidMetaType   = "invalid meta kind %s"
+
+	errCannotGet   = "cannot get the function input"
+	errCannotParse = "cannot parse the provided templates"
+)
+
+const (
+	annotationKeyCompositionResourceName = "crossplane.io/composition-resource-name"
+	annotationKeyReady                   = "meta.gotemplating.fn.crossplane.io/ready"
+
+	metaApiVersion = "meta.gotemplating.fn.crossplane.io/v1alpha1"
+)
+
 // RunFunction runs the Function.
 func (f *Function) RunFunction(_ context.Context, req *fnv1beta1.RunFunctionRequest) (*fnv1beta1.RunFunctionResponse, error) {
 	f.log.Info("Running Function", "tag", req.GetMeta().GetTag())
 
-	// This creates a new response to the supplied request. Note that Functions
-	// are run in a pipeline! Other Functions may have run before this one. If
-	// they did, response.To will copy their desired state from req to rsp. Be
-	// sure to pass through any desired state your Function is not concerned
-	// with unmodified.
 	rsp := response.To(req, response.DefaultTTL)
```

Comment:
@ezgidemirel after this line, could we forget about the request and always work on the response instead 🤔

Reply:
I mean read from response here: https://github.com/crossplane-contrib/function-go-templating/pull/1/files#diff-072fb060cc0b60d560101f306864ae72e9a08541205406d3566e13e05e023ae2R110-R117